ValetTests/ValetTests.swift

```diff
@@ -0,0 +1,742 @@
+//  You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import Foundation
+import XCTest
+
+
+extension NSError
+{
+    // The NSError.code -> VALMigrationError conversion is gross right now:
+    var valetMigrationError: VALMigrationError { return VALMigrationError(rawValue: UInt(self.code))! }
+}
+
+
+class TestValet: VALValet {}
+
+
+class ValetTests: XCTestCase
+{
+    static let identifier = "valet_testing"
+    let valet = VALValet(identifier: identifier, accessibility: .WhenUnlocked)!
+    let subclassValet = TestValet(identifier: identifier, accessibility: .WhenUnlocked)!
+    let key = "key"
+    let passcode = "topsecret"
+
+    // MARK: XCTestCase
+
+    override func setUp()
+    {
+        super.setUp()
+        valet.removeAllObjects()
+        subclassValet.removeAllObjects()
+    }
+
+    // MARK: Equality
+
+    func test_valetsWithSameConfiguration_areEqual()
+    {
+        let otherValet = VALValet(identifier: ValetTests.identifier, accessibility: .WhenUnlocked)
+        XCTAssert(otherValet == valet)
+        XCTAssert(otherValet === valet)
+    }
+
+    func test_differingSubclassesWithEquivalentConfiguration_areNotEqual()
+    {
+        XCTAssertFalse(valet == subclassValet)
+        XCTAssertFalse(valet === subclassValet)
+    }
+
+    func test_equivalentSubclassesWithEquivalentConfiguration_areEqual()
+    {
+        let secondSubclassValet = TestValet(identifier: ValetTests.identifier, accessibility: .WhenUnlocked)
+        XCTAssertNotNil(subclassValet)
+        XCTAssert(subclassValet == secondSubclassValet)
+        XCTAssert(subclassValet === secondSubclassValet)
+    }
+
+    func test_valetsWithDifferingIdentifier_areNotEqual()
+    {
+        let differingIdentifier = VALValet(identifier: "nope", accessibility: valet.accessibility)!
+        XCTAssertNotEqual(valet, differingIdentifier)
+    }
+
+    func test_valetsWithDifferingAccessibility_areNotEqual()
+    {
+        let differingAccessibility = VALValet(identifier: valet.identifier, accessibility: .Always)!
+        XCTAssertNotEqual(valet, differingAccessibility)
+    }
+
+    // MARK: canAccessKeychain
+
+    func test_canAccessKeychain()
+    {
+        XCTAssert(valet.canAccessKeychain())
+    }
+
+    func test_canAccessKeychain_Performance()
+    {
+        self.measureBlock {
+            self.valet.canAccessKeychain()
+        }
+    }
+
+    // MARK: containsObjectForKey
+
+    func test_containsObjectForKey()
+    {
+        XCTAssertFalse(valet.containsObjectForKey(key))
+
+        XCTAssert(valet.setString(passcode, forKey: key))
```

Comment:
nit: I really like using the specific asserts. e.g. `XCTAssertTrue(x, y)`, `XCTAssertEqual(x, y)`, etc.

Far from a blocker. But to me it feels better to be more specific. More obvious what's happening.